src/App.tsx

```diff
@@ -1,24 +1,52 @@
+/* eslint-disable max-classes-per-file */
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App: React.FC = () => {
-  window.setInterval(() => {
-    const date = new Date();
+type AppState = {
+  isClockVisible: boolean;
+  clockName: number;
+};
 
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
+class App extends React.Component<{}, AppState> {
+  buttonName = 'Hide';
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+  state: AppState = {
+    isClockVisible: true,
+    clockName: Math.round(Math.random() * 10),
+  };
+
+  setClock = (isClockVisible: boolean) => {
+    this.setState({ isClockVisible });
+    this.buttonName = this.state.isClockVisible ? 'Show' : 'Hide';
+  };
```

Comment:
Would be better to create function 'changeClockVisibility' without any arguments that set in state value '!isClockVisible'
And in render method use this value from state for displaying 'Show' or 'Hide' value for button